src/features/filter.ts

```diff
@@ -1,12 +1,25 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 
 const initialState = {
   query: '',
   status: 'all',
 };
 
-export const filterSlice = createSlice({
+const filterSlice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setQuery: (state, action: PayloadAction<string>) => ({
+      ...state,
+      query: action.payload,
```

Comment:
By the way,  Redux Toolkit's createSlice uses Immer under the hood, which allows you to mutate the state directly without needing to return a new object.
```suggestion
      state.query = action.payload;
    },
```